Exceptions/QueryParameterMismatchException.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Circle\DoctrineRestDriver\Exceptions;
+
+class QueryParameterMismatchException extends DoctrineRestDriverException
+{
+    /**
+     * @param $query
+     * @param $needle
+     */
+    public function __construct($query, $needle)
```

Comment:
Since this will be used in PHP 8+ we can declare `mixed $query, mixed $needle` unless its not supported by the parent ?

Reply:
Parent has no type hint - phpstorm is not showing any issues with adding them, but will leave it like it is - less risk of having a TypeError.